monai/bundle/utils.py

```diff
@@ -0,0 +1,169 @@
+        kwargs: other arguments for `json.load` or `yaml.load`, depends on file format.
+            for more details, please check:
+            https://docs.python.org/3/library/json.html#json.load.
+            https://pyyaml.org/wiki/PyYAMLDocumentation.
+
+    """
+    with open(filepath) as f:
+        if filepath.lower().endswith(".json"):
+            return json.load(f, **kwargs)
+        if filepath.lower().endswith((".yml", ".yaml")):
+            return yaml.safe_load(f, **kwargs)
+        raise ValueError("only support JSON or YAML config file so far.")
+
+
+def load_config_file_content(path: str, **kwargs):
+    """
+    Load part of the content from a config file with specified `id` in the path.
+    Suppprt JSON and YAML formats file.
+
+    Args:
+        path: path of target file to load, it can only load part of it appending target `id`
+            in the path with "#" mark. for example: `/data/config.json`, `/data/config.json#net#<args>`.
+        kwargs: other arguments for `json.load` or `yaml.load`, depends on file format.
+            for more details, please check:
+            https://docs.python.org/3/library/json.html#json.load.
+            https://pyyaml.org/wiki/PyYAMLDocumentation.
+
+    """
+    pattern = r"(json|yaml|yml)"
+    result = re.findall(pattern, path, re.IGNORECASE)
+    if len(result) != 1:
+        raise ValueError(f"path should only contain 1 file, but got: {path}.")
+
+    # split the path into filepath and target id of the content
+    paths = path.split(result[0])
+    parser = ConfigParser(config=load_config_file(paths[0] + result[0], **kwargs))
+    return parser[paths[1][1:] if paths[1] != "" else ""]
+
+
+def update_default_args(args: Optional[Union[str, Dict]] = None, **kwargs) -> Dict:
+    """
+    Update the `args` with the input `kwargs`.
+    For dict data, recursively update the content based on the keys.
+
+    Args:
+        args: source args to update.
+        kwargs: destination args to update.
+
+    """
+    args_: Dict = args if isinstance(args, dict) is None else {}  # type: ignore
+    if isinstance(args, str):
+        args_ = load_config_file_content(args)
+
+    # recursively update the default args with new args
+    for k, v in kwargs.items():
+        args_[k] = update_default_args(args_[k], **v) if isinstance(v, dict) and isinstance(args_.get(k), dict) else v
+    return args_
+
+
+def parse_id_value(pair: str) -> Tuple[str, Any]:
+    """
+    Parse the "id:value" pair string to `id` and `value`.
+    Will try to convert the data type of `value` from string to the correct type.
+
+    Args:
+        pair: input "id:value" pair to parse.
+
+    """
+    items = pair.split(":")
+    # remove blanks around id
+    id = items[0].strip()
+    value: Union[str, int, float, bool] = ""
+    if len(items) > 1:
+        # rejoin the rest, and remove blanks around value
+        value = ":".join(items[1:]).strip()
+
+    # try to convert the correct data type
```

Comment:
This if tree could be replaced by ast.literal_eval()?

Reply:
Totally removed these logic in the latest commit.
Let's rely on `fire` or other CLI tools directly.

Thanks.